api/src/middlewares/services/EmailService/CustomerEmailService.ts

```diff
@@ -0,0 +1,82 @@
+import { EmailService } from "./EmailService";
+import { FormField } from "../../../types/FormField";
+import { fieldsHashMap } from "./helpers";
+import * as additionalContexts from "./additionalContexts.json";
+import * as templates from "./templates";
+import config from "config";
+import { ApplicationError } from "../../../ApplicationError";
+import { NotifyMailModel, NotifyService } from "./NotifyService";
+
+const previousMarriageDocs = {
+  Divorced: "decree absolute",
+  "Dissolved civil partner": "final order",
+  Widowed: "late partner's death certificate",
+  "Surviving civil partner": "late partner's death certificate",
+  Annulled: "decree of nullity",
+};
+export class CustomerEmailService extends EmailService {
+  templateIds: {
+    standard: string;
+  };
+
+  declare provider: NotifyService;
+  constructor({ notifyService }) {
+    super("NOTIFY", notifyService);
+    if (!config.get("notifyTemplateStandard")) {
+      throw new ApplicationError("NOTIFY", "NO_TEMPLATE", 500);
+    }
+    this.templateIds = {
+      standard: config.get("notifyTemplateStandard"),
+    };
+  }
+
+  buildEmail(formData: FormField[], reference: string, paid: boolean): NotifyMailModel {
+    const defaultTemplate = templates.user.standard;
+    const fields = fieldsHashMap(formData);
+    const docsList = this.buildDocsList(fields, paid);
+    const country = fields["country"].answer;
+    const post = fields["post"]?.answer;
+    const personalisationValues = {
+      ...fields,
+      docsList,
+      paid,
+      reference,
+      ...additionalContexts[country as string],
+      ...(additionalContexts[post as string] ?? {}),
+    };
+    const personalisation = Object.keys(defaultTemplate).reduce((acc, curr) => {
+      return {
+        ...acc,
+        [curr]: personalisationValues[curr]?.answer ?? personalisationValues[curr],
+      };
+    }, {});
```

Comment:
For the values that don't have .answer, you can just put in the 2nd parameter so they don't need to be iterated over. (Or just not include them in the iterator at all?)


```.ts

const requiredKeys = Object.keys(defaultTemplate); // what happens to translationNeeded? will this get defaulted properly?.. where's it from?

 const personalisationsFromFields = requiredKeys.reduce((acc, curr) => {
      return {
        ...acc,
        [curr]: fields[curr].answer
      };
    }, {});

 const personalisations = {
    ...personalisationsFromFields,
    docsList,
    paid,
    reference,
    ...additionalContexts[country as string],
    ...(additionalContexts[post as string] ?? {}),
 }

```

or 

```.ts

 const personalisationsFromFields = requiredKeys.reduce((acc, curr) => {
      return {
        ...acc,
        [curr]: fields[curr].answer
      };
    }, 
    {
      docsList,
      paid,
      reference,
      ...additionalContexts[country as string],
      ...(additionalContexts[post as string] ?? {})
    });

   
 }

```